logging/resource.go

```diff
@@ -34,165 +33,172 @@ type commonResource struct{ *mrpb.MonitoredResource }
 
 func (r commonResource) set(l *Logger) { l.commonResource = r.MonitoredResource }
 
-var detectedResource struct {
-	pb   *mrpb.MonitoredResource
-	once sync.Once
+var detectedResource = struct {
+	pb    *mrpb.MonitoredResource
+	attrs internal.ResourceAtttributesGetter
+	once  *sync.Once
+}{
+	attrs: internal.ResourceAttributes(),
+	once:  new(sync.Once),
+}
+
+func metadataProjectID() (string, bool) {
```

Comment:
What happens if metadata file is corrupted? Is there a fallback for lookup in env variables?

Reply:
There is no metadata file. Please, see https://cloud.google.com/compute/docs/metadata/overview

Reply:
I believe adding this link to code would be a great reference!